libs/ConfigKit/include/ConfigComplex.h

```diff
@@ -0,0 +1,34 @@
+// Leka - LekaOS
+// Copyright 2023 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <filesystem>
+#include <span>
+
+namespace leka {
+
+template <std::size_t SIZE>
+struct ConfigComplex {
+  public:
+	explicit ConfigComplex(const std::filesystem::path &path, std::span<uint8_t> default_value = {})
```

Comment:
here, as this is the ctor, it's strange to want to pass something that might be bigger or smaller than the size of the config.

If you create an instance, you know what you are doing, so you know the size. I don't see a reason to pass two values when the config is expecting 3 -- in this case, if the 3 value in not important, then the value itself should say that

with this, you can actually change the ctor to:

```suggestion
	explicit ConfigComplex(const std::filesystem::path &path, std::array<uint8_t, SIZE> default_value = {})
```

this will allow you to also remove the `min_size + std::copy` and just assign the values to the array like so (simpler example):

```cpp
template<std::size_t SIZE>
struct Config {
std::array<int, SIZE> arr;
};

auto conf = Config<3> {1, 2, 3};
```

https://gcc.godbolt.org/z/a3b3vePqe